### PR TITLE
Feat: #367 Select and switch dapp account

### PR DESCRIPTION
**Pull request by hoailinh1210**
### Summary (Please recap what you changed or fixed)

image

image

- The user select more account will active first account in the list
- User can switch your account

### Checklist (If you won't pass this checklist please comment why)

-   [ ] I removed all commented or unnecessary code.
-   [ ] Provide the screenshots due to UI changed or bug fixed
-   [ ] My code has covered these test cases (please list down your tested cases):

### CODE REVIEW

File: client/src/components/hooks/useSelectWallet.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { useDispatch } from 'react-redux';
+import { setSelectedWallet } from '@cd/hooks/useServiceWorker';
+import { onBindingAuthInfo } from '@cd/actions/userActions';
+
+export const useSelectWallet = ({onSuccess} = {}) => {
+	const dispatch = useDispatch();
+	const selectWallet = useCallback(async (uid) => {
+		const result = await setSelectedWallet(uid);
+
+		const { publicKey: selectedPublicKey, userDetails } = result;
```
Comment (djhahe) [inline client/src/components/hooks/useSelectWallet.js:11]: no need to assign to `selectedPublicKey`

File: client/src/components/web-extension/DappConnection/index.jsx [hunk 2]
```diff
@@ -1,19 +1,49 @@
...
 import { useNavigate } from 'react-router-dom';
+import AccountItem from '@cd/common/AccountItem';
+import useGetWallets from '@cd/hooks/useGetWallets';
+import { useFormik } from 'formik';
+import { useSelectWallet } from '@cd/hooks/useSelectWallet';
+
 import './index.scss';
 
-const DappConnection = () => {
+const DappConnection = ({isUserExisting}) => {
```
Comment (djhahe) [inline client/src/components/web-extension/DappConnection/index.jsx:22]: pls check prettier config

**Review by djhahe — CHANGES_REQUESTED** [review]

File: client/src/components/web-extension/DappConnection/index.jsx [hunk 3]
```diff
@@ -26,41 +56,69 @@ const DappConnection = () => {
...
 			</div>
 			<div className="cd_we_connect_account_buttons">
-				<Button variant="primary" onClick={onOk}>
-					Yes
+				<Button variant="primary" type="submit" disabled={isDisabled}>
+					{isDisabled ? 'Select Account(s)' : 'Connect'}
 				</Button>
 
 				<Button variant="secondary" onClick={onCancel}>
 					No
```
Comment (djhahe) [inline client/src/components/web-extension/DappConnection/index.jsx:118]: Pls change the text to cancel

File: client/src/components/web-extension/DappConnection/index.jsx [hunk 4]
```diff
@@ -26,41 +56,69 @@ const DappConnection = () => {
...
+										selectedPublicKey={publicKey}
+									/>
+								</label>
+							);
+						})
+					}
+				</div>
 			</div>
 			<div className="cd_we_connect_account_buttons">
-				<Button variant="primary" onClick={onOk}>
```
Comment (djhahe) [inline client/src/components/web-extension/DappConnection/index.jsx:55]: let make 2 buttons horizontal in 1 line

Comment (hoailinh1210) [reply client/src/components/web-extension/DappConnection/index.jsx:55]: Updated

File: client/src/components/web-extension/DappConnection/index.jsx (hunk 2 above)
Comment (hoailinh1210) [reply client/src/components/web-extension/DappConnection/index.jsx]: Updated

File: client/src/components/hooks/useSelectWallet.js (hunk 1 above)
Comment (hoailinh1210) [reply client/src/components/hooks/useSelectWallet.js]: Updated

File: client/src/components/hooks/useGetWallets.js [hunk 5]
```diff
@@ -75,6 +75,11 @@ const useGetWallets = () => {
 		[loadWalletsFromStorage, loadBalanceWallets],
 	);
 
+	// const queryData = useInfiniteQuery({
```
[the remark this answers is not in the document]
Comment (hoailinh1210) [reply client/src/components/hooks/useGetWallets.js]: Updated

File: client/src/components/hooks/dapp/useGetCurrentConnectedUrl.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import { useQuery } from '@tanstack/react-query';
+import { getCurrentConnectedUrl } from '@cd/hooks/useServiceWorker';
+
+export const useGetCurrentConnectedUrl = () => {
+	return useQuery({
+		queryKey: ['dapp', 'currentConnectedUrl'],
+		queryFn: async () => {
+			return getCurrentConnectedUrl();
```
[the remark this answers is not in the document]
Comment (hoailinh1210) [reply client/src/components/hooks/dapp/useGetCurrentConnectedUrl.js]: Updated

File: client/template/extension/manifest.json [hunk 7]
```diff
@@ -11,9 +11,6 @@
 		"default_icon": "512.png",
 		"default_popup": "popup.html"
 	},
-	"content_security_policy": {
```
[the remark this answers is not in the document]
Comment (hoailinh1210) [reply client/template/extension/manifest.json]: hm, my bad just for testing ^^

File: client/src/components/web-extension/DappConnection/index.jsx (hunk 4 above)
Comment (djhahe) [reply client/src/components/web-extension/DappConnection/index.jsx:55]: main action should be on the right

**Comment by djhahe** [comment]
image
The current selected account should not have switch button

**Review by djhahe — APPROVED** [review]

**Pull request merged**